### ISSUE TITLE
Clang identifies use of reserved names in macros in utlist

### ISSUE THREAD

**Issue by aaronmdjones** [troydhanson/uthash#112]
Hello, me again!

I have pulled the latest utlist and uthash after issue #111 was resolved, and now I have more warnings to report.

```
lib/utils/utlist.h:95:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _SV(elt,list)
        ^
lib/utils/utlist.h:96:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _NEXT(elt,list,next) ((elt)->next)
        ^
lib/utils/utlist.h:97:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _NEXTASGN(elt,list,to,next) ((elt)->next)=(to)
        ^
lib/utils/utlist.h:99:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _PREVASGN(elt,list,to,prev) ((elt)->prev)=(to)
        ^
lib/utils/utlist.h:100:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _RS(list)
        ^
lib/utils/utlist.h:101:9: warning: macro name is a reserved identifier [-Wreserved-id-macro]
#define _CASTASGN(a,b) (a)=(b)
        ^
6 warnings generated.
```

The problem is that identifiers with a leading underscore are marked reserved by the language specification.

**Comment by Quuxplusone** [comment]
I've been putting this one off for a while now. ;) The bigger issue here is "utlist has a lot of helper macros that aren't properly namespaced", the biggest offenders being `DECLTYPE` and `LDECLTYPE` (whose semantics are subtly different — one substitutes for a type-specifier and the other for a type-cast — and neither of which is equivalent to standard C++ `decltype`) but also these underscored macros.

I propose to rename all of the un-namespaced helper macros to names beginning with `UT_` (or possibly even `UT_PORTABLE_` for these six). I would also like to rename and harmonize `DECLTYPE` and `LDECLTYPE` to `UTHASH_DECLTYPE` in the next release; that's the bit that I've been procrastinating on, but will take under consideration now that someone besides me has noticed them. ;)

The tricky bit with `DECLTYPE` is that it's a possible customization point for the user (i.e. someone might conceivably be compiling with `-DDECLTYPE=foo`), but (A) that doesn't apply to the helper macros you're talking about, and (B) I should take that risk anyway.

**Comment by Quuxplusone** [comment]
Should be fixed once I apply #115; @aaronmdjones please take a look! :)

**Comment by aaronmdjones** [comment]
No warnings. :)
